lib/evaluation/filterShouldExistIfPredicateEvaluationHandler.ts

```diff
@@ -0,0 +1,23 @@
+import { Field, InputDescriptor, PresentationDefinition } from '@sphereon/pe-models';
+
+import { Checked } from '../ConstraintUtils';
+
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+
+export class FilterShouldExistIfPredicateEvaluationHandler extends AbstractEvaluationHandler {
+
+  // This incorrect format of input_descriptor is a deal-breaker for us, and therefore we throw exception for it
+  public handle(pd: PresentationDefinition, p: any, result: Map<InputDescriptor, Map<any, Checked>>): void {
+    // HERE we process the predicate part of the PD
+    for (let i = 0; i < pd.input_descriptors.length; i++) {
+      const inputDescriptor: InputDescriptor = pd.input_descriptors[i];
+      for (let j = 0; j < inputDescriptor.constraints.fields.length; j++) {
```

Comment:
input_descriptor.constraints and input_descriptor.constraints.fields are not mandatory, thus an exception will be thrown if the input descriptor does not contain these properties .